### PR TITLE
Minting doesn't need special treatment after all

### LINKED ISSUE #263: Extra signer UTxO when minting
### Summary

When building a transaction where we mint tokens and pay them to a pkh, at the outputs there’s a UTxO with only minAda as its value that has no reason to exist. Seeing the balance of such a transaction seems that it comes from the UTxO where the transaction deducts the fees.

### Steps to reproduce the behavior

Try in the emulator an operation like
```
{-# INLINABLE mkPolicy #-}
mkPolicy :: () -> ScriptContext -> Bool
mkPolicy _ _ = True

coinMintingPolicy :: Scripts.MintingPolicy
coinMintingPolicy = mkMintingPolicyScript
    $$(PlutusTx.compile [|| Scripts.wrapMintingPolicy mkPolicy ||])

{-# INLINABLE coinCS #-}
coinCS :: CurrencySymbol
coinCS = scriptCurrencySymbol coinMintingPolicy

coinTokenName :: TokenName
coinTokenName = "coinToken"

mintingOperation
    :: forall w s
    .  Contract w s Text ()
mintingOperation = do
    pkh <- Contract.ownPaymentPubKeyHash

    let val = assetClassValue (Value.assetClass coinCS coinTokenName) 200

	lookups = Constraints.mintingPolicy coinMintingPolicy
        tx      = Constraints.mustMintValue val
               <> Constraints.mustPayToPubKey pkh (val <> minAda)

    ledgerTx <- submitTxConstraintsWith @Any lookups tx
```
That mints any amount of a coin and pays to a pkh. The outputs of a transaction like this always have an extra UTxO with _only_ minAda that belongs to the pubkey which pays the fees.

If we paid the minted tokens to a different pkh than the one paying fees the problem remains.

### Actual Result

We printed ledgerTx, a CardanoTx value that represents the above transaction. Their txOutputs and txMint fields looks like this
```
txOutputs = [
  TxOut {
    txOutAddress = Address {
      addressCredential = PubKeyCredential 80a4f45b56b88d1139da23bc4c3c75ec6d32943c087f250b86193ca7,
      addressStakingCredential = Nothing
    },
    txOutValue = Value (Map [(,Map [(\"\",45994238)])]),
    txOutDatumHash = Nothing
  },
  TxOut {
    txOutAddress = Address {
      addressCredential = PubKeyCredential 80a4f45b56b88d1139da23bc4c3c75ec6d32943c087f250b86193ca7,
      addressStakingCredential = Nothing
    },
    txOutValue = Value (Map [(,Map [(\"\",2000000)])]),
    txOutDatumHash = Nothing
  },
  TxOut {
    txOutAddress = Address {
      addressCredential = PubKeyCredential 80a4f45b56b88d1139da23bc4c3c75ec6d32943c087f250b86193ca7,
      addressStakingCredential = Nothing
    },
    txOutValue = Value (
      Map [
        (,Map [(\"\",2000000)]),
        (ddfaa51476c09e1534c81ad0a4c6c44276683727cacecae737e47dbb,Map [(\"coinToken\",200)])
      ]
    ),
    txOutDatumHash = Nothing}],
    txMint = Value (Map [(ddfaa51476c09e1534c81ad0a4c6c44276683727cacecae737e47dbb,Map [(\"coinToken\",200)])])
```

### Expected Result

That UTxO with only minAda should not exist - those minAda are “inside” the UTxO that pays the fees.

### Describe the approach you would take to fix this

_No response_

### System info

OS: Ubuntu
Version: 20.04
Plutus: c330fc60d0cd174743c95de71e5c404b5008f186

### PR DESCRIPTION
**Pull request by sjoerdvisscher**
Fixes #263 

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-contract/src/Wallet/Emulator/Wallet.hs [hunk 1]
```diff
@@ -405,52 +404,23 @@ handleBalanceTx utxo UnbalancedTx{unBalancedTxTx} = do
 -- | Adjust the left and right balance of an unbalanced 'Tx' with the missing
 -- lovelace considering the minimum lovelace per transaction output constraint
 -- from the Cardano blockchain.
-adjustBalanceWithMissingLovelace ::
```
Comment (silky) [inline plutus-contract/src/Wallet/Emulator/Wallet.hs:408]: 🥳

Comment (koslambrou) [reply plutus-contract/src/Wallet/Emulator/Wallet.hs:408]: Nice refactoring!

File: plutus-contract/test/Spec/Balancing.hs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        mkTx lookups constraints = either (error . show) id $ Constraints.mkTx @Void lookups constraints
+
+        mintingOperation :: Contract [Int] EmptySchema ContractError ()
+        mintingOperation = do
+            pkh <- Con.ownPaymentPubKeyHash
+
+            let val = vL 200
+                lookups = Constraints.mintingPolicy coinMintingPolicy
+                constraints = Constraints.mustMintValue val
+                    <> Constraints.mustPayToPubKey pkh (val <> Ada.toValue Ledger.minAdaTxOut)
```
[the remark this answers is not in the document]
Comment (sjoerdvisscher) [reply plutus-contract/test/Spec/Balancing.hs:119]: Yes, this is consistent with the other Constraints right?

[the remark this answers is not in the document]
Comment (koslambrou) [reply plutus-contract/test/Spec/Balancing.hs:119]: Yes it seems consistent! Thanks :)

**Pull request merged**